From cfa5f6d676d7bef8daee51562e61954c51607220 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Sat, 5 Dec 2020 20:42:49 +0530
Subject: [PATCH 19/40] Revert "ANDROID: x86, build: allow LTO_CLANG and
 THINLTO to be selected"

This reverts commit 6ca045ab401a466e9a5d52e5419a81753cdf3449.
---
 arch/x86/Kconfig  | 2 --
 arch/x86/Makefile | 5 -----
 2 files changed, 7 deletions(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index 4d77e6072f47..1c4c705cda2f 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -90,8 +90,6 @@ config X86
 	select ARCH_SUPPORTS_ACPI
 	select ARCH_SUPPORTS_ATOMIC_RMW
 	select ARCH_SUPPORTS_NUMA_BALANCING	if X86_64
-	select ARCH_SUPPORTS_LTO_CLANG		if X86_64
-	select ARCH_SUPPORTS_THINLTO		if X86_64
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_QUEUED_RWLOCKS
 	select ARCH_USE_QUEUED_SPINLOCKS
diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 3bafa7ce52af..94df0868804b 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -209,11 +209,6 @@ ifdef CONFIG_X86_64
 KBUILD_LDFLAGS += $(call ld-option, -z max-page-size=0x200000)
 endif
 
-ifdef CONFIG_LTO_CLANG
-KBUILD_LDFLAGS	+= -plugin-opt=-code-model=kernel \
-		   -plugin-opt=-stack-alignment=$(if $(CONFIG_X86_32),4,8)
-endif
-
 # Workaround for a gcc prelease that unfortunately was shipped in a suse release
 KBUILD_CFLAGS += -Wno-sign-compare
 #
-- 
2.17.1

